sakai-evaluation-impl/src/java/org/sakaiproject/evaluation/dao/EvaluationDaoImpl.java

```diff
@@ -2023,6 +2033,37 @@ public int selectConsolidatedEmailRecipients(boolean useAvailableEmailSent, Date
     	return count;
 	}
 
+    /*
+     * (non-Javadoc)
+     * @see org.sakaiproject.evaluation.dao.EvaluationDao#getAllSiteIDsMatchingSectionTitle(java.lang.String)
+     */
+    public Set<String> getAllSiteIDsMatchingSectionTitle( String sectionTitleWithWildcards )
+    {
+        String sql = String.format( SQL_SELECT_SITE_IDS_MATCHING_SECTION_TITLE, sectionTitleWithWildcards );
+        sql = sql + SQL_SELECT_SITE_IDS_MATCHING_SECTION_TITLE_MODIFIER;
+
+        Session session = getSessionFactory().openSession();
+        SQLQuery query = session.createSQLQuery( sql );
+        Set<String> results = new HashSet<>( query.list() );
+        session.close();
+        return results;
+    }
```

Comment:
I just refactored this code with success. I didn't realize that the SQLQuery object also has the `setParameter()` methods on it, which provides query sanitization. I tested it out locally and it does appear to be working.

The new code is as follows (ammendment in the other commit unfortunately, I can try to move it if need be):

```
    public Set<String> getAllSiteIDsMatchingSectionTitle( String sectionTitleWithWildcards )
    {
        Session session = getSessionFactory().openSession();
        SQLQuery query = session.createSQLQuery( SQL_SELECT_SITE_IDS_MATCHING_SECTION_TITLE );
        query.setParameter( "title", sectionTitleWithWildcards );
        Set<String> results = new HashSet<>( query.list() );
        session.close();
        return results;
    }

    public Set<String> getAllSiteIDsMatchingSiteTitle( String siteTitleWithWildcards )
    {
        Session session = getSessionFactory().openSession();
        SQLQuery query = session.createSQLQuery( SQL_SELECT_SITE_IDS_MATCHING_SITE_TITLE );
        query.setParameter( "title", siteTitleWithWildcards );
        Set<String> results = new HashSet<>( query.list() );
        session.close();
        return results;
    }
```